cpp/src/arrow/compute/function.cc

```diff
@@ -215,9 +215,15 @@ Result<Datum> Function::Execute(const std::vector<Datum>& args,
   }
 
   util::tracing::Span span;
-  START_SPAN(span, name(),
-             {{"detail", options ? options->ToString() : "<NULLPTR>"},
-              {"function.kind", kind()}});
+  START_SPAN(
+      span, name(),
+      {{"function.args", std::accumulate(std::begin(args), std::end(args), std::string(),
+                                         [](std::string& ss, const Datum& d) {
+                                           return ss.empty() ? d.ToString()
+                                                             : ss + "," + d.ToString();
+                                         })},
+       {"function.options", options ? options->ToString() : "<NULLPTR>"},
+       {"function.kind", kind()}});
```

Comment:
nit, but is the kind useful? We'll know from the definition and/or from context (e.g. GroupByNode vs ProjectNode)

Reply:
Same with the args - that'll give us datum types (array vs scalar), but I'm not sure what kind of analysis we'd do on that. (That said if collecting is cheap we might as well just always collect.)

Reply:
(On reflection, if we do need to group by kind, it'd be much easier to have it as an attribute instead of having to look it up in a registry, so kind seems okay to me. I'm not as sure about args.)

Reply:
Yes this is a trade-off for almost all attributes that are collected. I'll remove the args and keep the kind. We can always add/modify/remove attributes later.